### PR TITLE
Add support for custom overflow handlers

### PR DESCRIPTION
**Pull request by deongroenewald**
The existing implementation will destroy the request on any overflow errors which will abrubply terminate the connection without returning a response to the consumer.

Much like we did in #1 when we originally forked the library, this PR adds the options for adding custom handler functions when overflow errors are triggered so that the calling framework can handle the error more gracefully and return a meaningful response to the consumer.

## Links
- [data-api PR](https://github.com/lever/data-api/pull/1162)

### CODE REVIEW

File: lib/multipart.js [hunk 1]
```diff
@@ -422,8 +422,10 @@ Parser.handle = function(req, res, next, options) {
     , files = {};
 
   parser.on('error', function(err) {
-    req.destroy();
-    next(err);
+    if(err.hasCustomHandler !== true) {
```
Comment (marmstrong-lever) [inline lib/multipart.js:425]: (linting) missing space after `if`

File: lib/multipart.js [hunk 2]
```diff
@@ -441,8 +443,25 @@ Parser.handle = function(req, res, next, options) {
...
+      err = new Error('Multipart disk overflow')
+      if (typeof options.handleMultipartDiskOverflow === "function") {
+        options.handleMultipartDiskOverflow(err, req, res, next);
+        err.hasCustomHandler = true;
+        throw err;
+      }
+    } else if (this.written > limit) {
+      err = new Error('Multipart field size overflow');
+      if (typeof options.handleMultipartMaxFieldsSizeOverflow === "function") {
+        options.handleMultipartMaxFieldsSizeOverflow(err, req, res,next );
```
Comment (marmstrong-lever) [inline lib/multipart.js:457]: (linting) spacing around `next`

File: lib/multipart.js [hunk 3]
```diff
@@ -441,8 +443,25 @@ Parser.handle = function(req, res, next, options) {
...
+        err.hasCustomHandler = true;
+        throw err;
+      }
+    }
+
+    if(err) {
```
Comment (marmstrong-lever) [inline lib/multipart.js:463]: (linting) missing space after `if`

File: lib/multipart.js [hunk 4]
```diff
@@ -441,8 +443,25 @@ Parser.handle = function(req, res, next, options) {
   });
 
   parser.on('data', function() {
-    if (this.writtenDisk > diskLimit || this.written > limit) {
-      this.emit('error', new Error('Overflow.'));
+    var err = null;
+    if (this.writtenDisk > diskLimit) {
+      err = new Error('Multipart disk overflow')
```
Comment (marmstrong-lever) [inline lib/multipart.js:448]: (linting) missing `;`

File: lib/multipart.js [hunk 5]
```diff
@@ -441,8 +443,25 @@ Parser.handle = function(req, res, next, options) {
...
+      if (typeof options.handleMultipartDiskOverflow === "function") {
+        options.handleMultipartDiskOverflow(err, req, res, next);
+        err.hasCustomHandler = true;
+        throw err;
```
Comment (marmstrong-lever) [inline lib/multipart.js:452]: Just for my own knowledge- what's the reason why we want to throw the error here rather than emit an error event like below?

**Review by marmstrong-lever — APPROVED** [review]
LGTM :tada: :tada: just a couple minor linting notes!

File: lib/multipart.js (hunk 5 above)
Comment (deongroenewald) [reply lib/multipart.js:452]: Emitting an error doesn't stop it from continuing to process the body, whereas throwing here will abort the entire process.

**Pull request merged**